modules/client/src/internal/client/methods.ts

```diff
@@ -1116,408 +1101,163 @@ export class ClientMethods extends ClientCore implements IClientMethods {
-    }
-
-    // get the prepared setup id
-    const preparedSetupId = getPreparedSetupId(
-      decodedApplyUpdateActionParams,
-      PreparationType.UPDATE,
-    );
-    // get plugin preparation
-    type W = { pluginPreparation: SubgraphPluginUpdatePreparation };
-    const { pluginPreparation } = await this.graphql.request<W>({
-      query: QueryPluginPreparations,
-      params: { where: { preparedSetupId } },
-      name: "pluginPreparation",
+  public async isPluginUpdate(
+    proposalId: string,
+  ): Promise<boolean> {
+    const name = "iproposal";
+    const query = QueryIProposal;
+    type T = { iproposal: SubgraphIProposal };
+    const { iproposal } = await this.graphql.request<T>({
+      query,
+      params: { id: proposalId.toLowerCase() },
+      name,
     });
-    if (pluginPreparation) {
-      // get the metadata of the plugin repo
-      // for the release and build specified
-      const release = pluginRepo.releases.find((
-        release: SubgraphPluginRepoRelease,
-      ) =>
-        release.release === decodedApplyUpdateActionParams.versionTag.release
-      );
-      const build = release?.builds.find((
-        build: { build: number; metadata: string },
-      ) => build.build === decodedApplyUpdateActionParams.versionTag.build);
-      const metadataCid = build?.metadata;
-
-      // fetch the metadata
-      const metadata = await this.ipfs.fetchString(metadataCid!);
-      const metadataJson = JSON.parse(metadata) as PluginRepoBuildMetadata;
-      // get the update abi for the specified build
-      const updateAbi = metadataJson.pluginSetup
-        .prepareUpdate[decodedApplyUpdateActionParams.versionTag.build]?.inputs;
-      if (updateAbi) {
-        // if the abi exists try to decode the data
-        try {
-          if (
-            decodedApplyUpdateActionParams.initData.length > 0 &&
-            updateAbi.length === 0
-          ) {
-            throw new Error();
-          }
-          // if the decode does not throw an error the data is valid
-          defaultAbiCoder.decode(
-            getNamedTypesFromMetadata(updateAbi),
-            decodedApplyUpdateActionParams.initData,
-          );
-        } catch {
-          // if the decode throws an error the data is invalid
-          causes.push(
-            PluginUpdateProposalInValidityCause.INVALID_DATA,
-          );
-        }
-      } else {
-        causes.push(
-          PluginUpdateProposalInValidityCause.INVALID_PLUGIN_REPO_METADATA,
-        );
-      }
-    } else {
-      causes.push(
-        PluginUpdateProposalInValidityCause.MISSING_PLUGIN_PREPARATION,
-      );
+    if (!iproposal) {
+      return false;
     }
-    return causes;
+    const subgraphActions = iproposal.actions;
+    let actions = toDaoActions(subgraphActions);
+    const classifiedActions = classifyProposalActions(actions);
+    return isPluginUpdateAction(classifiedActions) ||
+      isPluginUpdateActionWithRootPermission(classifiedActions);
```

Comment:
```suggestion
      isPluginUpdateActionBlockWithRootPermission(classifiedActions);
```

Reply:
This name is getting longer and longer. However, this is ok as long as it is precise and describes exactly what the function is doing.
Generally, the need to give long names is a hint towards not having the right code structure. The length of names can often be reduced by organizing it differently.
E.g., splitting code into smaller files with meaningful file names / module scope (i.e., not having a single `methods.ts` containing everything, but categorizing file by their purpose (e.g., `proposal_analysis.ts`)) or putting them into classes/objects (e.g., a Proposal class the defines blocks/segments).